### ISSUE TITLE
Plotting animations on Ubuntu 22.04 do not work

### ISSUE THREAD

**Issue by gustavowl** [hiperwalk/hiperwalk#6]
Plotting animations on Ubuntu 22.04 do not work.
To test, run the following code.

```
import hiperwalk as hpw
g = hpw.Cycle(11)
qw = hpw.ContinuousTime(graph=g, gamma=0.35)
states = qw.simulate(time=(10, 1), initial_state=qw.ket(5))
probs = qw.probability_distribution(states)
hpw.plot_probability_distribution(probs, animate=True, show=True)
```

**Comment by gmgall** [comment]
This doesn't seem to be related with the Ubuntu version but with the GTK version.

It works if you set the 3.0 version outside of the `hiperwalk/plot/_animation.py` file.

I tried the test code provided above with a `hiperwalk/plot/__init__.py` with the following content:

```python
# importing everything under the hood of the
# hiperwalk.plot package
from gi import require_version
require_version('Gtk', '3.0')
from ._plot import *
from ._animation import *
```

And it seems to work. Could anyone else test to confirm?

**Comment by gustavowl** [comment]
> This doesn't seem to be related with the Ubuntu version but with the GTK version.
> 
> It works if you set the 3.0 version outside of the `hiperwalk/plot/_animation.py` file.
> 
> I tried the test code provided above with a `hiperwalk/plot/__init__.py` with the following content:
> 
> ```python
> # importing everything under the hood of the
> # hiperwalk.plot package
> from gi import require_version
> require_version('Gtk', '3.0')
> from ._plot import *
> from ._animation import *
> ```
> 
> And it seems to work. Could anyone else test to confirm?

Important: the content of the ```hiperwalk/plot/_animation.py``` file was not changed.

**Comment by gmgall** [comment]
@gustavowl confirmed that this change makes it work on Ubuntu 22.04.

**Comment by gmgall** [comment]
358fc53 force the use of Gtk 3.0.

 `require_version()` raises

```python
ValueError: Namespace Gtk is already loaded with version 4.0
```

in `hiperwalk/plot/_animation.py`. 

This **doesn't** happen if `require_version()` is used in `hiperwalk/plot/__init__.py`.

We need to investigate why and evaluate if `__init.py__` is the best place to set the Gtk version, therefore this issue will be kept open for now.

**Comment by gmgall** [comment]
TL;DR: looks like 358fc53a03e5051dee1c4ebe8ac3d0f13d6c8692 can be safely undone but I would like to confirm with someone else.

---

358fc53a03e5051dee1c4ebe8ac3d0f13d6c8692 introduced the side effect of making `gi` a dependency to `import` the package. It's possible to see that when trying to `import` hiperwalk in an environment like a JupyterLite REPL.

358fc53a03e5051dee1c4ebe8ac3d0f13d6c8692 was a quick fix to the problem of hiperwalk raising an `ValueError` complaining about the loaded GTK version in Ubuntu 22.04. See lines 237 to 241 in the  [`hiperwalk/plot/_animation.py`](https://github.com/hiperwalk/hiperwalk/blob/master/hiperwalk/plot/_animation.py#L239) file.

But I'm not able to reproduce the problem anymore. I tried to test it as follows:

1. installed Ubuntu 22.04 in a fresh virtual machine
2. downloaded hiperwalk code from repository
3. installed `python3-pip` via `apt`
4. installed `numpy`, `scipy`, `networkx` and `matplotlib` via `pip`
5. deleted following lines from `hiperwalk/plot/__init__.py`
```python
from gi import require_version
require_version('Gtk', '3.0')
```
6. imported hiperwalk in a Python session without errors

**It looks like 358fc53a03e5051dee1c4ebe8ac3d0f13d6c8692 can be safely undone** but I wasn't able to find out exactly what changed from 358fc53a03e5051dee1c4ebe8ac3d0f13d6c8692 to today. This would allow at least `import`ing hiperwalk in JupyterLite and everything would still work in Ubuntu 22.04

**Comment by gustavowl** [comment]
Before making a PR, assert that the following works for Ubuntu 22.04.
1. Undo  [`358fc53`](https://github.com/hiperwalk/hiperwalk/commit/358fc53a03e5051dee1c4ebe8ac3d0f13d6c8692)
2. In the terminal, execute the code
```python
import hiperwalk as hpw
import numpy as np

dim = 41
grid = hpw.Grid(dim, diagonal=True)
center = np.array([dim // 2, dim // 2])
dtqw = hpw.Coined(grid, shift='persistent', coin='grover')
psi0 = dtqw.state([0.5, (center, center + (1, 1))],
                  [-0.5, (center, center + (1, -1))],
                  [-0.5, (center, center + (-1, 1))],
                  [0.5, (center, center + (-1, -1))])
states = dtqw.simulate(time=(dim // 2, 1), initial_state=psi0, hpc=False)
prob = dtqw.probability_distribution(states)
hpw.plot_probability_distribution(prob, graph=grid, rescale=True, animate=True)
```
3. An animation shall be displayed.

**Comment by gmgall** [comment]
The code above helped me see my confusion. I was testing without specifying `animate=True`, resulting in the code that uses `gi` never being called. :sweat_smile: 

Something like 358fc53a03e5051dee1c4ebe8ac3d0f13d6c8692 is still necessary because `gi.require_version()` raises a `ValueError` when called inside a method.

In the top of a file, next of other `import` statements, it works fine for some reason.
